src/api/productAPI.ts

```diff
@@ -30,33 +41,62 @@ export async function getCategoryProductList({
   sort,
   page,
   size,
-  company,
-  switchType,
+  companies,
+  switchTypes,
   minPrice,
   maxPrice,
 }: GetCategoryListParams): Promise<ProductListResponse> {
   try {
-    const queryParams = new URLSearchParams({
-      keyword: encodeURIComponent(keyword),
-      sort: encodeURIComponent(sort as string),
-      page: encodeURIComponent(page as string),
-      size: encodeURIComponent(size),
-      ...(company && { company: encodeURIComponent(company as string) }),
-      ...(switchType && { switchType: encodeURIComponent(switchType as string) }),
-      ...(minPrice && { minPrice: encodeURIComponent(minPrice as string) }),
-      ...(maxPrice && { maxPrice: encodeURIComponent(maxPrice as string) }),
-    }).toString();
-
-    const response = await fetch(`${BASE_URL}/api/v1/product/category/${keyword}?${queryParams}`);
+    const queryParams: Record<string, string> = {
+      keyword,
+      sort,
+      page,
+      size,
+    };
+
+    if (companies) queryParams.companies = companies;
+    if (switchTypes) queryParams.switchTypes = switchTypes;
+    if (minPrice) queryParams.minPrice = minPrice;
+    if (maxPrice) queryParams.maxPrice = maxPrice;
```

Comment:
companies~maxPrice 가 undefined 가 아닌 값이 온다면 이렇게 쓰는 게 나을 거 같긴한데 (null, 0 등)
undefined 이면 구냥 넣어줘도 어짜피 companies : undefined 이면 안넣어준 거랑 똑같아서 그냥 한 번에 처리해도 될 거 같아요!
```suggestion
   const queryParams: Record<string, string> = {
      keyword,
      sort,
      page,
      size,
      companies,
      switchTypes,
      minPrice,
      maxPrice
      
    };
```